category_encoders/one_hot.py

```diff
@@ -193,8 +193,10 @@ def get_dummies(X_in, cols=None):
 
         bin_cols = []
         for col in cols:
-            classes = [x for x in set(X[col].values.tolist()) if np.isfinite(x)]
-            for class_ in classes:
+            col_tuples = [class_map['mapping'] for class_map in ordinal_encoder.mapping if class_map['col'] == col][0]
+            fit_classes = set(col_val[1] for col_val in col_tuples)
+            input_classes = set(x for x in set(X[col].values.tolist()) if np.isfinite(x))
+            for class_ in (fit_classes | input_classes):
```

Comment:
If a new class is passed in to transform and handle_unknown == 'impute' then the -1 class is automatically added when `self.ordinal_encoder.transform(X)` is called. That is why I take the union of fit_classes and input_classes (input_classes includes the -1 class, in this specific case).

Otherwise, I could change it to:
```
fit_classes = set(col_val[1] for col_val in col_tuples)
if (self.handle_unknown == 'impute') & (-1 in X[col].unique()):
  fit_classes.update({-1})
for class_ in fit_classes:
```
That way the -1 class is only added when it's present. Either way, a new class column is only added if everything passes correctly through the ordinal encoder.